prov/util/src/util_coll.c

```diff
@@ -773,12 +773,14 @@ static int util_coll_find_local_rank(struct fid_ep *ep, struct util_coll_mc *col
 	size_t addrlen;
 	char *addr;
 	int ret, mem;
+	fi_addr_t  my_addr;
+        int i;
 
 	addrlen = sizeof(mem);
 	addr = (char *) &mem;
 
 	ret = fi_getname(&ep->fid, addr, &addrlen);
-	if (ret != -FI_ETOOSMALL) {
+	if (ret != -FI_ETOOSMALL && sizeof(size_t) >= addrlen) {
```

Comment:
The flow through here does not look correct.  We should just set addrlen = 0 and addr = NULL, get the correct size, and fall through so that lookup below can be done.

Please update the commit message with a signed-off-by line, and change the commit title to:
```
util/coll: Find correct local collective rank